marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/entities/search/addDocumentMetadataToSearchResults.sjs

```diff
@@ -0,0 +1,101 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+declareUpdate();
+const entitySearchLib = require("/data-hub/5/entities/entity-search-lib.sjs");
+const test = require("/test/test-helper.xqy");
+
+function verifyMetadataForDocumentWithAllMetadata() {
+  let response = {
+    "snippet-format": "snippet",
+    "total": 1,
+    "results": [
+      {
+        "index": 1,
+        "uri": "/content/sally.json",
+      }
+    ]
+  };
+
+  let expectedSources = [{"name":"loadData"}, {"name":"someOtherName"}];
+
+  xdmp.invokeFunction(function() {
+    declareUpdate();
+    xdmp.documentSetMetadata("/content/sally.json",
+        {
+          "datahubCreatedByStep": "map-step",
+          "datahubCreatedInFlow": "CurateCustomerJSON",
+          "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+        });
+  });
+  entitySearchLib.addDocumentMetadataToSearchResults(response);
+  return[
+    test.assertEqual("CurateCustomerJSON", response.results[0].metadata.Flow),
+    test.assertEqual("map-step", response.results[0].metadata.Step),
+    test.assertEqual("2020-10-08T15:14:28.772612-07:00", response.results[0].metadata.createdOn),
+    test.assertEqual(expectedSources, response.results[0].metadata.sources)
+  ];
+}
+
+function verifyMetadataForDocumentWithoutMetadata() {
+  let response = {
+    "snippet-format": "snippet",
+    "total": 1,
+    "results": [
+      {
+        "index": 1,
+        "uri": "/content/jane.json",
+      }
+    ]
+  };
+  entitySearchLib.addDocumentMetadataToSearchResults(response);
+  return[
+      test.assertEqual(0, Object.keys(response.results[0].metadata).length)
+  ];
+}
+
+function verifyMetadataForDocumentWithPartialMetadata() {
+  let response = {
+    "snippet-format": "snippet",
+    "total": 1,
+    "results": [
+      {
+        "index": 1,
+        "uri": "/content/tim.json",
+      }
+    ]
+  };
+
+  xdmp.invokeFunction(function() {
+    declareUpdate();
+    xdmp.documentSetMetadata("/content/tim.json",
+        {
+          "datahubCreatedByStep": "map-step"
+        });
+  });
+  entitySearchLib.addDocumentMetadataToSearchResults(response);
+  return[
+    test.assertEqual("", response.results[0].metadata.Flow),
```

Comment:
Why empty string instead of not setting the property at all? This is often a problematic approach for dates and dateTimes, where "" is not a valid value. I think it's best to just not include the property at all - that makes it crystal clear that it doesn't exist for this record.